src/main/java/seedu/address/logic/commands/AddCommand.java

```diff
@@ -61,10 +61,7 @@ public CommandResult executeUndoableCommand() throws CommandException {
         try {
             model.addParcel(toAdd);
             model.maintainSorted();
-            if (model.hasSelected()) {
```

Comment:
try to make maintainSorted called in model rather than in the command itself. Like perhaps put it in the internal logic of addParcel. Then model.reselect(toAdd) can also be in addParcel().